pixeltable/functions/globals.py

```diff
@@ -168,6 +168,11 @@ def _(val: sql.ColumnElement) -> Optional[sql.ColumnElement]:
     return sql.sql.func.avg(val)
 
 
+def map(expr: exprs.Expr, fn: Callable[[exprs.Expr], Any]) -> exprs.Expr:
+    target_expr = exprs.Expr.from_object(fn(exprs.json_path.RELATIVE_PATH_ROOT))
```

Comment:
This should generate a user-relatable error if the expression cannot be generated.

I'm a bit worried that a lambda looks very general, but really needs to be fairly specific.

Reply:
Ok. Note that `@pxt.query` has the identical problem (arguably much worse in that case, since it's a always full `def`, rather than a little lambda). So does `@expr_udf`, of course, but we're already moving away from that one.

Reply:
Agreed, and that's why expr_udf is mostly hidden away. In this case, I think you can create a somewhat user-friendly error message.

Reply:
I don't think this needs to be elaborate, just somewhat more informative than some obscure stack trace.

Also, it would be good to add a test for that.

Reply:
Done.